src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1964,8 +1983,6 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
         }
     }
 
-    buildInternalRegisterUses();
```

Comment:
This was a latent bug.

Previously, we define the use for the internal register, and then afterwards build the definition of the destination. There is only one use for the internal, therefore the destination def is free to reuse the same register.

This bug was never seen in practice because all HWIntrisnsics that had immediate values had a vector register for the destination. The internal register is always a scalar register, and so cannot be reused.

SaturatingDecrement is the first HWIntrisnsics with both an immediate and a scalar destination.

Fix is to simply move the internal use after the destination definition.

Reply:
We always want all the uses to be present before the definition. 

> SaturatingDecrement is the first HWIntrisnsics with both an immediate and a scalar destination.

Yes, and hence as we discussed yesterday, the proper fix would be to mark the internal register as delay free *under certain circumstances* such as this.

Reply:
Got it - I didn't get that it was the internal register that should be delayed. Makes sense now. And fixed.

Reply:
And with that change all the stress tests are working.